### PR TITLE
Fixed error handling

### PR DESCRIPTION
**Pull request by keiya01**
I fixed error handling in useFieldArray because control's error log was not invoked. When control and methods.control is not defined, JavaScript throw TypeError. Therefore I fixed to throw our error before JavaScript throw TypeError.

If this PR is correct, I will fix other part.

### CODE REVIEW

File: src/useFieldArray.ts [hunk 1]
```diff
@@ -49,6 +50,17 @@ export const useFieldArray = <
   keyName = 'id' as TKeyName,
 }: UseFieldArrayOptions<TKeyName, TControl>) => {
   const methods = useFormContext();
+
+  if (process.env.NODE_ENV !== 'production') {
+    if (!control && !methods) {
+      throw new RHFError('📋 useFieldArray is missing `control` prop.');
```
Comment (bluebill1049) [inline src/useFieldArray.ts:56]: shall we keep it simple just throw `new Error`?

Comment (keiya01) [reply src/useFieldArray.ts:56]: Sure🙇‍♂️

**Comment by keiya01** [comment]
@bluebill1049 
Could you review this PR when you have a time?

**Comment by bluebill1049** [comment]
> @bluebill1049
> Could you review this PR when you have a time?

Will do 🙏

**Comment by keiya01** [comment]
Thank you very much 🙏

**Comment by keiya01** [comment]
Probably yarn server is down now 🙄

File: src/controller.test.tsx [hunk 2]
```diff
@@ -501,74 +501,289 @@ describe('Controller', () => {
...
+
+      expect(console.warn).not.toBeCalled();
+
+      process.env.NODE_ENV = env;
+
+      // @ts-ignore
+      console.warn.mockRestore();
+    });
+
+    it('should warn the user when defaultValue is missing with useFieldArray in development environment', () => {
```
Comment (bluebill1049) [inline src/controller.test.tsx:705]: ty for those tests!

File: src/controller.tsx [hunk 3]
```diff
@@ -29,6 +29,11 @@ const Controller = <
   ...rest
 }: ControllerProps<TAs, TControl>) => {
   const methods = useFormContext();
+
+  if (process.env.NODE_ENV !== 'production' && !control && !methods) {
```
Comment (bluebill1049) [inline src/controller.tsx:33]: 👍

File: src/useFieldArray.ts [hunk 4]
```diff
@@ -49,6 +49,17 @@ export const useFieldArray = <
   keyName = 'id' as TKeyName,
 }: UseFieldArrayOptions<TKeyName, TControl>) => {
   const methods = useFormContext();
+
+  if (process.env.NODE_ENV !== 'production') {
```
Comment (bluebill1049) [inline src/useFieldArray.ts:53]: 👍

File: src/useWatch.ts [hunk 5]
```diff
@@ -37,6 +37,17 @@ export function useWatch<TWatchFieldValues>({
   defaultValue,
 }: UseWatchOptions): TWatchFieldValues {
   const methods = useFormContext();
+
+  if (process.env.NODE_ENV !== 'production') {
```
Comment (bluebill1049) [inline src/useWatch.ts:41]: 👍

**Review by bluebill1049 — APPROVED** [review]
LGTM!

**Comment by bluebill1049** [comment]
ty @keiya01

**Comment by keiya01** [comment]
@bluebill1049 thank you for your review🙏

**Pull request merged**